src/server/editorServices.ts

```diff
@@ -541,174 +576,162 @@ namespace ts.server {
             // We don't respond to the second one.
             if (fileName && !isSupportedSourceFileName(fileName, project.getCompilerOptions(), this.hostConfiguration.extraFileExtensions)) {
+                this.logger.info(`Project: ${configFilename} Detected file add/remove of non supported extension: ${fileName}`);
                 return;
             }
 
-            this.logger.info(`Detected source file changes: ${fileName}`);
-            this.throttledOperations.schedule(
-                project.getConfigFilePath(),
-                /*delay*/250,
-                () => this.handleChangeInSourceFileForConfiguredProject(project, fileName));
+            const configFileSpecs = project.configFileSpecs;
+            const result = getFileNamesFromConfigSpecs(configFileSpecs, getDirectoryPath(configFilename), project.getCompilerOptions(), project.getCachedServerHost(), this.hostConfiguration.extraFileExtensions);
+            const errors = project.getAllProjectErrors();
+            const isErrorNoInputFiles = (error: Diagnostic) => error.code === Diagnostics.No_inputs_were_found_in_config_file_0_Specified_include_paths_were_1_and_exclude_paths_were_2.code;
+            if (result.fileNames.length !== 0) {
+                filterMutate(errors, error => !isErrorNoInputFiles(error));
+            }
+            else if (!configFileSpecs.filesSpecs && !some(errors, isErrorNoInputFiles)) {
+                errors.push(getErrorForNoInputFiles(configFileSpecs, configFilename));
+            }
+            this.updateNonInferredProjectFiles(project, result.fileNames, fileNamePropertyReader, /*clientFileName*/ undefined);
+            this.delayUpdateProjectGraphAndInferredProjectsRefresh(project);
         }
 
-        private handleChangeInSourceFileForConfiguredProject(project: ConfiguredProject, triggerFile: string) {
-            const { projectOptions, configFileErrors } = this.convertConfigFileContentToProjectOptions(project.getConfigFilePath());
-            this.reportConfigFileDiagnostics(project.getProjectName(), configFileErrors, triggerFile);
-
-            const newRootFiles = projectOptions.files.map((f => this.getCanonicalFileName(f)));
-            const currentRootFiles = project.getRootFiles().map((f => this.getCanonicalFileName(f)));
-
-            // We check if the project file list has changed. If so, we update the project.
-            if (!arrayIsEqualTo(currentRootFiles.sort(), newRootFiles.sort())) {
-                // For configured projects, the change is made outside the tsconfig file, and
-                // it is not likely to affect the project for other files opened by the client. We can
-                // just update the current project.
-
-                this.logger.info("Updating configured project");
-                this.updateConfiguredProject(project);
+        private onConfigChangedForConfiguredProject(project: ConfiguredProject, eventKind: FileWatcherEventKind) {

```

Comment:
Same here with `eventKind` -- only compared against one possible kind, so might as well be a boolean.